test/ui/users/restrictions/RestrictionsFormContainer.test.js

```diff
@@ -1,38 +1,77 @@
-import { formValueSelector } from 'redux-form';
-import { mapStateToProps, mapDispatchToProps } from 'ui/users/restrictions/RestrictionsFormContainer';
+import React from 'react';
+import * as reactRedux from 'react-redux';
 
-const dispatchMock = jest.fn();
+import { renderWithState } from 'test/testUtils';
+import RestrictionsFormContainer from 'ui/users/restrictions/RestrictionsFormContainer';
+import RestrictionsForm from 'ui/users/restrictions/RestrictionsForm';
 
-jest.unmock('react-intl');
+jest.unmock('react-redux');
 
-jest.mock('redux-form', () => ({
-  formValueSelector: jest.fn(() => jest.fn(() => false)),
-  reduxForm: jest.fn(() => jest.fn(ui => ui)),
-}));
+const useDispatchSpy = jest.spyOn(reactRedux, 'useDispatch');
+const mockDispatch = jest.fn();
+useDispatchSpy.mockReturnValue(mockDispatch);
 
-jest.mock('state/loading/selectors', () => ({
-  getLoading: jest.fn().mockReturnValue({}),
+jest.mock('state/user-settings/actions', () => ({
+  fetchUserSettings: jest.fn(() => ({ type: 'fetchUserSettings_test' })),
+  updateUserSettings: jest.fn(payload => ({ type: 'updateUserSettings_test', payload })),
 }));
 
+jest.mock('ui/users/restrictions/RestrictionsForm', () => jest.fn(mockProps => (
+  <div>
+    <button onClick={mockProps.onMount}>onMount</button>
+    <button onClick={() => mockProps.onSubmit({})}>onSubmit</button>
+  </div>
```

Comment:
what are the purpose of these stuff?

Reply:
That's one way of testing a container component without having to fully render the child component since the child component already has its own tests. Basically, it's simulating the component's events via button clicks.

Reply:
ok makes sense then. Thanks